### PR TITLE
feat(lib): add popper API

### PR DESCRIPTION
**Pull request by geromegrignon**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

A popper needs to be defined with the directive on the host element.

Issue Number: #7 

## What is the new behavior?

The popper can be added programatically.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/helipopper/src/lib/helipopper.service.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export class HelipopperService {
+  constructor(
+    private appRef: ApplicationRef,
+    private zone: NgZone,
+    private resolver: ComponentFactoryResolver,
+    private hostInjector: Injector,
+    @Inject(HELIPOPPER_CONFIG) private config: HelipopperConfig
+  ) {}
+
+  open(host: ElementRef, helipopper: string | TemplateRef<any>, options?: HelipopperOptions) {
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/helipopper.service.ts:26]: It could be also a component and we should allow passing a custom Injector.

**Review by NetanelBasal — COMMENTED** [review]
LGTM. You can move forward.

File: projects/ngneat/helipopper/src/lib/helipopper.service.ts (hunk 1 above)
Comment (geromegrignon) [reply projects/ngneat/helipopper/src/lib/helipopper.service.ts:26]: sure. About the injector part i'll update the actual README as i created the option interface based on it and the @Input injector is missing

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
-import { Component } from '@angular/core';
```
Comment (NetanelBasal) [inline src/app/app.component.ts:1]: Can you add an example of using a component and nested one more popper inside it, please?

**Comment by geromegrignon** [comment]
> Sorry for the confusion. I meant, popper that in its template opens another popper - nested poppers.

Do you have a use case in mind? As i can find one with the current API, i can't find one fitting with the new API (which would justify to attach the popper manually).

**Comment by NetanelBasal** [comment]
Yes. https://atomiks.github.io/tippyjs/#nesting

**Comment by geromegrignon** [comment]
> Yes. https://atomiks.github.io/tippyjs/#nesting

I updated the example, tell me if it fits your vision.

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
-import { FormBuilder } from '@angular/forms';
+import { AfterViewInit, Component, ElementRef, ViewChild } from '@angular/core';
+import { FormBuilder, FormControl } from '@angular/forms';
 import { ExampleComponent } from './example/example.component';
 import { interval } from 'rxjs';
 import { finalize } from 'rxjs/operators';
+import { HelipopperDirective, HelipopperService } from '@ngneat/helipopper';
+import { MenuComponent } from './menu/menu.component';
```
Comment (NetanelBasal) [inline src/app/app.component.ts:7]: Why I can't see the menu component?

Comment (geromegrignon) [reply src/app/app.component.ts:7]: sorry, some files were in another changelist, i committed them.

File: projects/ngneat/helipopper/src/lib/helipopper.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      this.appRef,
+      this.zone,
+      this.resolver,
+      this.hostInjector,
+      this.config
+    );
+
+    directive.helipopper = helipopper;
+
+    directive.helipopperOptions = options?.helipopperOptions || {};
```
Comment (NetanelBasal) [inline projects/ngneat/helipopper/src/lib/helipopper.service.ts:40]: Can we please make one single source of truth for the default options we use both here and in the directive? Such as placement, etc.

Comment (geromegrignon) [reply projects/ngneat/helipopper/src/lib/helipopper.service.ts]: done, i updated the new interface too by removing 'helipopper' in the properties wording as it's redundant with the interface name.

**Comment by NetanelBasal** [comment]
Thanks! I'll check it with my use case, and merge it.

**Pull request merged**